Translate CoffeeScript Files To JavaScript (ES2015)

joefitzgerald opened the pull request:
This does what it says on the box. Zero functional changes; just translation of CoffeeScript to JavaScript ES2015. Each non-spec file will be converted, tested with unmodified spec files, then the corresponding spec file(s) will be converted, tested, and pushed. When CI passes, the next non-spec file will go through the same process, until all files are converted.

- [x] Enable `standard` linter
- [x] Fix existing lint issues in `symbol-store.js`
- [x] `autocomplete-manager.coffee`
- [x] `autocomplete-manager-async-spec.coffee`
- [x] `autocomplete-manager-autosave-spec.coffee`
- [x] `autocomplete-manager-integration-spec.coffee`
- [x] `autocomplete-manager-undo-spec.coffee`
- [x] `fuzzy-provider.coffee`
- [x] `fuzzy-provider-spec.coffee`
- [x] `main.js`
- [x] `main-spec.coffee`
- [x] `provider-api-legacy-spec.coffee`
- [x] `provider-api-spec.coffee`
- [x] `private-symbols.coffee`
- [x] `provider-manager.coffee`
- [x] `provider-manager-spec.coffee`
- [x] `provider-metadata.coffee`
- [x] `ref-counted-token-list.coffee`
- [x] `ref-counted-token-list-spec.coffee`
- [x] `scope-helpers.coffee`
- [x] `snippet-parser.coffee`
- [x] `spec-helper.coffee`
- [x] `suggestion-list-element.coffee`
- [x] `suggestion-list-element-spec.coffee`
- [x] `suggestion-list.coffee`
- [x] `symbol-provider.coffee`
- [x] `symbol-provider-spec.coffee`
- [x] `symbol-store.js`
- [x] `symbol-store-spec.coffee`
- [x] `type-helpers.coffee`
- [x] `unicode-helpers.coffee`
- [x] Incorporate changes from #779 
- [x] Incorporate changes from #782
- [x] Incorporate changes from #678

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+    editorSubscriptions.add(onDidTokenizeProvider.onDidTokenize(() => {
+      return this.buildWordListOnNextTick(editor)
+    }))
```
This block can be changed to
```js
editorSubscriptions.add(onDidTokenizeProvider.onDidTokenize(() => this.buildWordListOnNextTick(editor)))
```

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      return this.buildWordListOnNextTick(editor)
+    }))
+    editorSubscriptions.add(editor.onDidDestroy(() => {
+      let index = this.getWatchedEditorIndex(editor)
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      let bufferSubscriptions = new CompositeDisposable()
+      if (bufferSupportsStopChanging()) {
+        bufferSubscriptions.add(buffer.onDidStopChanging(({changes}) => {
+          let editors = this.watchedBuffers.get(buffer)
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+        }))
+      } else {
+        bufferSubscriptions.add(buffer.onDidChange(({oldRange, newRange}) => {
+          let editors = this.watchedBuffers.get(buffer)
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+        bufferSubscriptions.add(buffer.onDidChange(({oldRange, newRange}) => {
+          let editors = this.watchedBuffers.get(buffer)
+          if (editors && editors.length && editors[0]) {
+            let { start } = oldRange
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+          let editors = this.watchedBuffers.get(buffer)
+          if (editors && editors.length && editors[0]) {
+            let { start } = oldRange
+            let oldExtent = oldRange.getExtent()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+          if (editors && editors.length && editors[0]) {
+            let { start } = oldRange
+            let oldExtent = oldRange.getExtent()
+            let newExtent = newRange.getExtent()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+  }
+
+  getWatchedEditorIndex (editor) {
+    let editors = this.watchedBuffers.get(editor.getBuffer())
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+  buildConfig (scopeDescriptor) {
+    this.config = {}
+    let legacyCompletions = this.settingsForScopeDescriptor(scopeDescriptor, 'editor.completions')
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    allConfigEntries.reverse()
+
+    for (let i = 0; i < legacyCompletions.length; i++) {
+      let {value} = legacyCompletions[i]
```
Change `let` to `const`. Also add some whitespace inside the brackets to be consistent with other, similar code?

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+    let addedConfigEntry = false
+    for (let j = 0; j < allConfigEntries.length; j++) {
+      let {value} = allConfigEntries[j]
```
Change `let` to `const`. Same whitespace remark as earlier.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+  }
+
+  addConfigEntry (config) {
+    for (let type in config) {
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+  addConfigEntry (config) {
+    for (let type in config) {
+      let options = config[type]
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      this.config[type].typePriority = options.typePriority != null ? options.typePriority : 1
+      this.config[type].wordRegex = this.wordRegex
+
+      let suggestions = this.sanitizeSuggestionsFromConfig(options.suggestions, type)
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    }
+
+    this.buildConfigIfScopeChanged(options)
+    let editor = options.editor
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+    this.buildConfigIfScopeChanged(options)
+    let editor = options.editor
+    let bufferPosition = options.bufferPosition
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    this.buildConfigIfScopeChanged(options)
+    let editor = options.editor
+    let bufferPosition = options.bufferPosition
+    let prefix = options.prefix
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    let prefix = options.prefix
+
+    let numberOfWordsMatchingPrefix = 1
+    let wordUnderCursor = this.wordAtBufferPosition(editor, bufferPosition)
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+    let numberOfWordsMatchingPrefix = 1
+    let wordUnderCursor = this.wordAtBufferPosition(editor, bufferPosition)
+    let iterable = editor.getCursors()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      if (word === wordUnderCursor) { numberOfWordsMatchingPrefix += 1 }
+    }
+
+    let buffers = this.includeCompletionsFromAllBuffers ? null : [this.editor.getBuffer()]
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      prefix, wordUnderCursor,
+      bufferPosition.row,
+      numberOfWordsMatchingPrefix
+    )
```
Is there a reason for this somewhat unorthodox formatting? This should be broken on every comma, right? Alternatively break once, and leave all parameters on a single line.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+  }
+
+  wordAtBufferPosition (editor, bufferPosition) {
+    let lineToPosition = editor.getTextInRange([[bufferPosition.row, 0], bufferPosition])
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      prefix = ''
+    }
+
+    let lineFromPosition = editor.getTextInRange([bufferPosition, [bufferPosition.row, Infinity]])
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+        return
+      }
+
+      let start = {row: 0, column: 0}
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+      }
+
+      let start = {row: 0, column: 0}
+      let oldExtent = {row: 0, column: 0}
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+
+      let start = {row: 0, column: 0}
+      let oldExtent = {row: 0, column: 0}
+      let newExtent = editor.getBuffer().getRange().getExtent()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    if (a === b) { return true }
+    if ((a == null) || (b == null)) { return false }
+
+    let arrayA = a.getScopesArray()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-provider.js:
```diff
@@ -0,0 +1,371 @@
+    if ((a == null) || (b == null)) { return false }
+
+    let arrayA = a.getScopesArray()
+    let arrayB = b.getScopesArray()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-store.js:
```diff
@@ -102,11 +102,11 @@ export default class SymbolStore {
 
-      let symbolsByLetter = new Map
+      let symbolsByLetter = new Map()
       let tokenIterator = tokenizedLine.getTokenIterator()
```
Change `let` to `const`.

thomasjo commented on lib/symbol-store.js:
```diff
@@ -102,11 +102,11 @@ export default class SymbolStore {
+      let symbolsByLetter = new Map()
       let tokenIterator = tokenizedLine.getTokenIterator()
       while (tokenIterator.next()) {
         let wordsWithinToken = tokenIterator.getText().match(this.wordRegex) || EMPTY_ARRAY
```
Change `let` to `const`.

thomasjo commented on lib/type-helpers.js:
```diff
@@ -0,0 +1,13 @@
+'use babel'
+
+let isFunction = value => isType(value, 'function')
```
Change `let` to `const`.

thomasjo commented on lib/type-helpers.js:
```diff
@@ -0,0 +1,13 @@
+
+let isFunction = value => isType(value, 'function')
+
+let isString = value => isType(value, 'string')
```
Change `let` to `const`.

thomasjo commented on lib/type-helpers.js:
```diff
@@ -0,0 +1,13 @@
+
+let isString = value => isType(value, 'string')
+
+var isType = function (value, typeName) {
```
Just make this a "real" function instead? `function isType (value, typeName) { `
I guess the same goes for the functions above, but at least here it avoids re-ordering and means we can drop the `var` specifier.

thomasjo commented on lib/type-helpers.js:
```diff
@@ -0,0 +1,13 @@
+let isString = value => isType(value, 'string')
+
+var isType = function (value, typeName) {
+  let t = typeof value
```
Change `let` to `const`.

as-cii commented:
@joefitzgerald just a quick heads-up: as we discussed I have merged #782, so there's going to be some merge conflict. Please, ping me if you need any help back porting the changes I made in CoffeeScript. Thanks!

joefitzgerald commented:
@as-cii I incorporated the changes from #782 in 5be853f0d6af4275f8cd41b292073b8da6988ac2; it's worth noting that all the builds for the intermediate commits were green prior to rebasing. I am inclined to merge #779 now to minimize further merge conflicts. You cool with that?

joefitzgerald commented:
@thomasjo I addressed your review comments; please review the whole PR now, as it is complete. Obviously we could improve the code further, but I don't want to focus on picking up every possible improvement, and perhaps address additional work (if significant) in subsequent PRs.

as-cii commented:
Thank you so much @joefitzgerald!!! ❤️ ❤️ ❤️
